spec/money_spec.rb

```diff
@@ -493,6 +493,22 @@
       ).to eq([Money.new(2600, 'JPY'), Money.new(475, 'JPY')])
     end
 
+    specify "#allocate_max_amounts legal computation with no currency objects" do
+      expect(
+        Money.new(3075, 'JPY').allocate_max_amounts([2600, 475]),
+      ).to eq([Money.new(2600, 'JPY'), Money.new(475, 'JPY')])
+
+      expect(
+        Money.new(3075, Money::NullCurrency.new).allocate_max_amounts([Money.new(2600, 'JPY'), Money.new(475, 'JPY')]),
+      ).to eq([Money.new(2600, 'JPY'), Money.new(475, 'JPY')])
```

Comment:
Nevermind seems it's not a problem, indeed both are converted to subunits using the same multiplier 👍